plugins/brain/__init__.py

```diff
@@ -2221,6 +2225,314 @@ def _inject_brain_secrets(ctx):
+    )
+    dataset = sample_collection._dataset
+    dataset.save_view(
+        "exact_duplicates_view", exact_duplicates_view, overwrite=True
+    )
+    dataset.save_view(
+        "representatives_of_exact_duplicates_view",
+        representatives_of_exact_duplicates_view,
+        overwrite=True,
+    )
+
+    return {
+        "num_unique_imgs_with_dups": len(duplicates_dict),
+        "num_total_dups": len(flat_duplicates),
+    }
+
+
+class DeduplicateExactDuplicates(foo.Operator):
+    @property
+    def config(self):
+        return foo.OperatorConfig(
+            name="deduplicate_exact_duplicates",
+            label="Deduplicate exact duplicates",
+            description="Remove all but one copy from each group of exact duplicates in the dataset",
+            light_icon="/assets/icon-light.svg",
+            dark_icon="/assets/icon-dark.svg",
+            dynamic=True,
+        )
+
+    def resolve_input(self, ctx: foo.ExecutionContext) -> types.Property:
+        inputs = types.Object()
+        form_view = types.View(
+            label="Deduplicate exact duplicates",
+            description="Deduplicate exact duplicates in the dataset",
+        )
+        return types.Property(inputs, view=form_view)
+
+    def execute(self, ctx: foo.ExecutionContext) -> None:
+
+        sample_collection = ctx.dataset
+        dataset = sample_collection._dataset
+
+        if "exact_duplicates_view" not in dataset.list_saved_views():
+            find_exact_duplicates(sample_collection)
+
+        exact_duplicates_view = dataset.load_saved_view(
+            "exact_duplicates_view"
+        )
+        representatives_of_exact_duplicates_view = dataset.load_saved_view(
+            "representatives_of_exact_duplicates_view"
+        )
+
+        remove_sample_ids = [
+            _id
+            for _id in exact_duplicates_view.values("id")
+            if not _id in representatives_of_exact_duplicates_view.values("id")
+        ]
+
+        dataset.delete_samples(remove_sample_ids)
+
+        dataset.delete_saved_view("exact_duplicates_view")
+        dataset.delete_saved_view("representatives_of_exact_duplicates_view")
+
+        if not ctx.delegated:
+            ctx.trigger("reload_dataset")
+
+
+def find_approximate_duplicates(
+    sample_collection: focc.SampleCollection, threshold: float = None
+) -> dict[str, int]:
+    """
+    Find approximate duplicate images in the given sample collection using
+    embeddings.
+    Args:
+        sample_collection: The FiftyOne sample collection to search for
+            approximate duplicates.
+        threshold: The distance threshold for determining approximate duplicates.
+    Returns:
+        A dictionary containing the number of unique images with at least one
+        approximate duplicate and the total number of approximate duplicates found.
+    """
+
+    dataset = sample_collection._dataset
+    index = fob.compute_near_duplicates(dataset, threshold=threshold)
```

Comment:
For consistency with other operators in `@voxel51/brain`, it would be ideal if all of the [optional kwargs](https://docs.voxel51.com/api/fiftyone.brain.html#fiftyone.brain.compute_near_duplicates
) of `compute_near_duplicates()` were exposed to the user as optional parameters in `resolve_input()` so that you could pass them here:

You can achieve this by [copying the same pattern](https://github.com/voxel51/fiftyone-plugins/blob/e7adfebc9869dccf64fde89ca4b7f5274fc8ccd2/plugins/brain/__init__.py#L1241-L1262) used by `@voxel51/brain/compute_uniqueness`.